### DISCUSSION TITLE
GitHub Internal Grading System -- future features?

### DISCUSSION THREAD

**Discussion by catcoding02** [community/community#114484] [GitHub Education]
### Hi everyone,

I am an instructor for an introductory data science course where students practice cleaning and wrangling data and creating maps. I cannot implement autograding like other coding instructors because of the creative nature of student assignments. We have a homegrown learning management system that relies on extracting comments on a student's assignment that include keywords to indicate a student's score, which is then sent to Google Sheets (which stores student grades in worksheets) and Slack (which handles notifications), and I was wondering if GitHub plans to add features offering ways for instructors to directly grade work in GitHub and allow students to be notified of grades and view their grades, or if anyone has suggestions for more efficient ways to build grading out of GitHub classroom than my current set-up.

**Comment by IsmaelMousa** [comment]
Hello **catcoding**, Check this [GitHub Education](https://education.github.com/teachers).

**Reply by catcoding02** [reply]
Hi @IsmaelMousa, I was wondering if there was a particular page on the linked website you would recommend? It just looks like general information about GitHub Education and Classroom.

**Reply by IsmaelMousa** [reply]
@catcoding02 actually I think the most popular way is to fill the students grades in README.md file as a table,
and for real-time updates you need to get the students grades API and determine what the information you want to be filled in the table.

I hope my answer was helpful.

**Reply by catcoding02** [reply]
What is the students grades API?

**Reply by IsmaelMousa** [reply]
In the context of your purpose, you'll need to utilize the Google Sheets API to access the students' grades data. The Google Sheets API is a set of tools provided by Google that allows software applications to interact with Google Sheets programmatically. Essentially, it provides a way for your application to read and manipulate data stored in Google Sheets.

To get started, you'll need to obtain the URL of the Google Sheets document containing the students' grades. With this URL, you can then use the Google Sheets API to fetch the relevant data.

It's important to note that when accessing data through the Google Sheets API, authentication is required to ensure the security of the data. You'll need to set up authentication credentials, such as API keys or OAuth tokens, to authenticate your requests to the API.

For a step-by-step guide on how to use the Google Sheets API to read data from Google Sheets, I recommend watching this tutorial video: [Google Sheets API Tutorial](https://www.youtube.com/watch?v=3wC-SCdJK2c). It provides a clear demonstration of how to fetch data from Google Sheets using the API.

**Reply by catcoding02** [reply]
I’m sorry if my original post wasn’t clear—we already are doing all of that. That’s the only way we are able to make connections between what is in GitHub and what is in Google Sheets. I was asking if there was a better way people knew of besides juggling several different APIs and having to deal with slow response times and quotas.

**Reply by IsmaelMousa** [reply]
Got it!, I think you will got the answer if you checked the GitHub Education very well and understanding the abilities that it can offer for you.

BTW I see this and I think it related to your question: [Auto-Grading Assignments with GitHub Classroom](https://www.youtube.com/watch?v=uHwYuZvys6k).

I hope you will find the solution, best of luck.

**Reply by catcoding02** [reply]
Thanks! I’ve looked pretty deeply and I think the features just aren’t there. Was hoping GitHub staff would see this post and respond if new features were in the works, but seems like I will have to further develop what we have already created. I definitely think this is a gap in the market that GitHub is missing out on…but not sure how in-demand an in-house LMS is for GitHub teachers.

**Reply by IsmaelMousa** [reply]
Unfortunately yes, but wait you can open an Issue (Feature Request) in GitHub and telling them about your feature, they are kindness and taking the recommendation from the users.

Here is the GitHub Org:  https://github.com/orgs/github
And here is a video that explain how to open Feature Request: https://www.youtube.com/watch?v=j1-xU6bmZZg
